R/wrangle_and_check.R

```diff
@@ -29,3 +29,58 @@ wrangle_and_check_sector_exposures_eq_cb <- function(sector_exposures, asset_typ
 
   return(valid_sector_exposures)
 }
+
+
+#' Wrangle and check PACTA results for asset types equity and bonds
+#'
+#' Function applies several filter steps to
+#' 1. restrict PACTA results to scenarios useable for stresstesting
+#' 1. to apply run/project related selections.
+#'
+#' Also consistency checks are run.
+#'
+#' @param pacta_results Results from PACTA analysis of type equity or bond.
+#' @param start_year First considered year in analysis.
+#' @param time_horizon Considered timefram for PACTA analysis.
+#' @param scenario_geography_filter Character. A vector of length 1 that
+#'   indicates which geographic scenario to apply in the analysis.
+#' @param scenarios_filter Vector holding baseline and shock scenario name.
+#' @param equity_market_filter Character. A vector of length 1 that
+#'   indicates which equity market to apply in the analysis.
+#'
+#' @return Wrangled `pacta_results.`
+#' @export
+wrangle_and_check_pacta_results_eq_cb <- function(pacta_results, start_year, time_horizon,
+                                                  scenario_geography_filter, scenarios_filter,
+                                                  equity_market_filter) {
+
+  wrangled_pacta_results <- pacta_results %>%
+    dplyr::filter(!is.na(.data$scenario)) %>%
+    check_scenario_settings(scenario_selections = allowed_scenarios_eq_cb) %>%
+    dplyr::filter(.data$scenario %in% allowed_scenarios_eq_cb) %>%
```

Comment:
this seems like an implicit argument? Or am I missing something here?

Reply:
u mean allowed_scenarios_eq_cb? This is available since it is configured in lookup.R I moved a couple of variables (sectors, technologies..) there to not overload the function interface, given the names are very specific I thinkt this should be no problem